tests/integration/test_matmul.py

```diff
@@ -147,19 +147,23 @@ def test_out_invalid_shape_DIVERGENCE(self):
         out = num.zeros(shape)
         num.matmul(A, B, out=out)
 
-    def test_out_invalid_dtype(self):
+    @pytest.mark.parametrize(
+        ("dtype", "out_dtype", "casting"),
+        ((None, np.int64, "same_kind"), (float, str, "safe")),
+        ids=("direct", "intermediate"),
+    )
+    def test_out_invalid_dtype(self, dtype, out_dtype, casting):
         expected_exc = TypeError
-        A_np = num.ones((3, 2, 4))
-        B_np = num.ones((3, 4, 3))
+        A_np = np.ones((3, 2, 4))
+        B_np = np.ones((3, 4, 3))
         A_num = num.ones((3, 2, 4))
         B_num = num.ones((3, 4, 3))
-        dtype = np.int64
-        out_np = np.zeros((3, 2, 3), dtype=dtype)
-        out_num = num.zeros((3, 2, 3), dtype=dtype)
+        out_np = np.zeros((3, 2, 3), dtype=out_dtype)
+        out_num = num.zeros((3, 2, 3), dtype=out_dtype)
```

Comment:
I think `empty` would suffice for "out" arrays (in general) but it's not a blocker to me